src/UnitTests/Projection/MapFromTest.cs

```diff
@@ -21,6 +21,18 @@ public void Should_not_fail()
         }
 
         [Fact]
+        public void Should_not_fail_Untyped()
+        {
+            var config = new MapperConfiguration(cfg =>
+            {
+                cfg.CreateMap<UserModel, UserDto>()
+                    .ForMember(dto => dto.FullName, opt => opt.MapFrom(src => src.LastName + " " + src.FirstName));
+            });
+
+            typeof(ArgumentNullException).ShouldNotBeThrownBy(() => config.ExpressionBuilder.GetMapExpression(typeof(UserModel), typeof(UserDto), null, null)); //no ArgumentNullException here
```

Comment:
Your decision, I changed the checks and now all the args for that method are null checked, obviously had to move the ``parameters ?? new Dictionary<string, object>();`` to a different place (namely ``SourceInjectedQueryProvider`` where the ?? for the ``MemberInfos`` is).
I still don't really like the logical difference between the typed and untyped api